### ISSUE TITLE
Incorrect pop in popModal (js/ui/popupMenu.js)

### ISSUE THREAD

**Issue by dalcde** [linuxmint/Cinnamon#1126]
How to reproduce:
Open the menu. Before it opens, press Alt-F2 to launch run dialog.

Result:
The run dialog loses keyboard focus and cannot be closed

Console log:

JS ERROR: !!!   Exception in callback for signal: open-state-changed
    JS ERROR: !!!     message = '"incorrect pop"'
    JS ERROR: !!!     lineNumber = '892'
    JS ERROR: !!!     fileName = '"/usr/share/cinnamon/js/ui/main.js"'
    JS ERROR: !!!     stack = '"popModal([object _private_St_BoxLayout])@/usr/share/cinnamon/js/ui/main.js:892
()@/usr/share/cinnamon/js/ui/popupMenu.js:1837
([object Object],false)@/usr/share/cinnamon/js/ui/popupMenu.js:1883
_emit("open-state-changed",false)@/usr/share/gjs-1.0/signals.js:124
(true)@/usr/share/cinnamon/js/ui/popupMenu.js:1246
()@/usr/share/cinnamon/js/ui/popupMenu.js:2033
([object _private_Clutter_Stage],[object _private_GObject_ParamSpec])@/usr/share/cinnamon/js/ui/popupMenu.js:1953
pushModal([object _private_St_Group],1301810)@/usr/share/cinnamon/js/ui/main.js:865
((void 0))@/usr/share/cinnamon/js/ui/modalDialog.js:257
()@/usr/share/cinnamon/js/ui/modalDialog.js:207
()@/usr/share/cinnamon/js/ui/runDialog.js:387
([object _private_Meta_Display],[object _private_Meta_Screen],null,[object _private_Meta_KeyBinding])@/usr/share/cinnamon/js/ui/main.js:141
